### ISSUE TITLE
Error: Permission denied to access property "then"

### ISSUE THREAD

**Issue by welwood08** [NiklasGollenstede/stop-fingerprinting#7]
I've encountered a site attempting to use `navigator.mediaDevices.enumerateDevices().then(...)` which fails with the console message `Error: Permission denied to access property "then"`. Naturally, the whole site breaks at this point. Adding an exception for the site works when I tested in Firefox 53.0a2 with e10s on.

Some comments in https://bugzilla.mozilla.org/show_bug.cgi?id=1120862 appear to be relevant from glancing at the code in content/src/fake/media-devices.js - it looks like maybe the empty array is not being created on the right global? Maybe something like `new ucw.Array()` instead?

**Comment by NiklasGollenstede** [comment]
Yeah, that's an incredibly ugly kind of error - I still haven't found the perfect way to fix that.

If you like, you can try the beta version I just uploaded to [AMO](https://addons.mozilla.org/en-US/firefox/addon/stop-fingerprinting/versions/beta).
It uses an entirely different technique to inject its code (with `new Function('string')()`. While that fixes any permission problems when accessing the resulting objects, it completely fails on sites that disallow `unsafe-eval` in their CSP (which are not that many, but some sites do that, which is generally a good thing).
I hope I will find the time to produce a proper fix soon.
